astore/server/astore/retrieve.go

```diff
@@ -76,15 +79,14 @@ func (s *Server) Retrieve(ctx context.Context, req *astore.RetrieveRequest) (*as
 			return nil, status.Errorf(codes.InvalidArgument, "Invalid path - %s", err)
 		}
 	} else {
-		query = datastore.NewQuery(KindArtifact).Order("-Created")
+		query = datastore.NewQuery(KindArtifact)
 	}
 	query = query.Limit(1)
 
 	if req.Uid != "" {
 		query = query.Filter("Uid = ", req.Uid)
 	}
-
-	tags := []string{"latest"}
```

Comment:
so, if "no list of tags" is specified at all today, it is assumed to be "latest"

if a list of tags is specified (and this list of tags can be empty?) then the list is used.

Shouldn't the client then specify an empty non-nil list of tags when uid is supplied?

(doesn't removing the lines here affect the default behavior?)

Reply:
I think well-behaved clients looking to download a package version are specifying the version either by UID or tag; failure to do so should be an error (otherwise people can download by path, rely on default behavior, never think about what version they should be getting, and get broken unexpectedly when `latest` is bad)

If we want to have defaults, I think defaults should be boosted as close to the user as possible (i.e. client can set tag default if appropriate) and the server should be as dumb as possible.